src/BlockTypes/ProductGalleryPager.php

```diff
@@ -37,96 +39,128 @@ protected function get_block_type_uses_context() {
-	 */
-	private function render_pager( $pager_display_mode ) {
-		switch ( $pager_display_mode ) {
-			case 'dots':
-				return $this->render_dots_pager();
-			case 'digits':
-				return $this->render_digits_pager();
-			case 'off':
-				return null;
-			default:
-				return $this->render_dots_pager();
+			$html = $this->render_pager( $product_gallery_images_ids, $pager_display_mode, $number_of_thumbnails );
+
+			return sprintf(
+				'<div %1$s>
+					%2$s
+				</div>',
+				$wrapper_attributes,
+				$html
+			);
 		}
 	}
 
 	/**
-	 * Renders the digits pager HTML.
+	 * Renders the pager for the product gallery.
 	 *
-	 * @return string The rendered digits pager HTML.
+	 * @param  array  $product_gallery_images_ids An array of image IDs for the product gallery.
+	 * @param  string $pager_display_mode         The display mode for the pager.
+	 * @param  int    $number_of_thumbnails       The number of thumbnails to display in the pager.
+	 * @return string|null The rendered pager HTML, or null if the pager should not be displayed.
 	 */
-	private function render_digits_pager() {
-		return sprintf(
-			'<ul class="wp-block-woocommerce-product-gallery-pager__pager">
-				<li class="wp-block-woocommerce-product-gallery__pager-item is-active">1</li>
-				<li class="wp-block-woocommerce-product-gallery__pager-item">2</li>
-				<li class="wp-block-woocommerce-product-gallery__pager-item">3</li>
-				<li class="wp-block-woocommerce-product-gallery__pager-item">4</li>
-			</ul>'
-		);
+	private function render_pager( $product_gallery_images_ids, $pager_display_mode, $number_of_thumbnails ) {
+		if ( $number_of_thumbnails < 2 || 'off' === $pager_display_mode ) {
+			return null;
+		}
+
+		return $this->render_pager_pages( $product_gallery_images_ids, $number_of_thumbnails, $pager_display_mode );
 	}
 
 	/**
-	 * Renders the dots pager HTML.
+	 * Renders the pager pages for the product gallery.
 	 *
-	 * @return string The rendered dots pager HTML.
+	 * @param  array  $product_gallery_images_ids An array of image IDs for the product gallery.
+	 * @param  int    $number_of_thumbnails The number of thumbnails to display in the pager.
+	 * @param  string $pager_display_mode The display mode for the pager. Defaults to 'dots'.
+	 * @return string The rendered pager pages HTML.
 	 */
-	private function render_dots_pager() {
+	private function render_pager_pages( $product_gallery_images_ids, $number_of_thumbnails, $pager_display_mode = 'dots' ) {
+		$html = '';
+
+		foreach ( $product_gallery_images_ids as $key => $product_gallery_image_id ) {
+			if ( $key >= $number_of_thumbnails ) {
+				break;
+			}
+
+			$is_first_pager_item = 0 === $key;
+			$pager_item          = sprintf(
+				'<li class="wp-block-woocommerce-product-gallery__pager-item %2$s">%1$s</li>',
+				'dots' === $pager_display_mode ? $this->get_dot_icon( $is_first_pager_item ) : $key + 1,
+				$is_first_pager_item ? 'is-active' : ''
```

Comment:
`is-active` is too much generic. Let's use a class with `wc` prefix. https://github.com/woocommerce/woocommerce-blocks/blob/73151afd5929b747fb8bd98c37337e22a830ee2e/docs/contributors/coding-guidelines.md/#L15-L18

Reply:
I've tried to change the `is-active` to `.wc-block-woocommerce-product-gallery__pager-item--is-active`; however, it breaks Interactivity API when using with the `data-wc-class` directive, for example:
```
$p->set_attribute(
	'data-wc-class--wc-block-woocommerce-product-gallery__pager-item--is-active',
	'selectors.woocommerce.isSelected'
);
```

It seems that this directive does not accept the `__` or `--` combination. The only way I've managed to make it work was to change the classname to `wc-block-woocommerce-product-gallery-pager-item-is-active` but it goes against the guidelines that you shared. Any suggestions?

Reply:
Good point! I would update with `wc-block-woocommerce-product-gallery-pager-item-is-active`and adding a comment that explains why. What do you think?

Reply:
I followed your suggestion to use `wc-block-woocommerce-product-gallery-pager-item-is-active` and left a comment there explaining why

Reply:
We fixed that in Core in this PR, but we haven't updated Woo's codebase yet:

- https://github.com/WordPress/gutenberg/pull/53337

Feel free to replace the old reg exp with the new one yourselves 🙂

Reply:
Hey @luisherranz,
Thank you for sharing the solution for this with us! I just opened a PR that implements this: https://github.com/woocommerce/woocommerce-blocks/pull/11034